### DISCUSSION TITLE
Missing error detection for some functions

### DISCUSSION THREAD

**Discussion by ericemmrich** [vimeo/psalm#9940] [Q&A]
Hey folks,

during a recent PHP 8 upgrade, we noticed that Psalm seems to be missing error detection for some functions. 

In particular, we set up a Psalm config to only report `InvalidScalarArgument` errors:

```xml
<issueHandlers>
  <!-- Report only the following code issues -->
  <InvalidScalarArgument errorLevel="error" />
```

And created a basic test script:

```php
<?php

chr('2');
abs('2');
floor('2');
```

When running Psalm on this test script, only the `floor()` function reports an error, even though all of these functions require a valid `int|float`:

```
$ ./vendor/bin/psalm --no-cache --config=./psalm-81.xml test.php

ERROR: InvalidScalarArgument - test.php:9:7 - Argument 1 of floor expects float, but '2' provided (see https://psalm.dev/012)
floor('2');


------------------------------
1 errors found
------------------------------
```

Are we missing something?

**Edit:** Running the test script on https://psalm.dev, at least `chr()` gets reported as well. `abs()` is still not reported, though:

![image](https://github.com/vimeo/psalm/assets/12201765/a948a613-b935-4314-b973-c2c03655ef98)

**Comment by orklah** [comment]
Before PHP 8, PHP used to specifically support numeric-string for abs: https://www.php.net/manual/en/function.abs.php#refsect1-function.abs-changelog

It has led to a specific signature in Psalm's internals: https://github.com/vimeo/psalm/blob/e3f589d79e6ced49336cec5016dd9abce9c310ba/dictionaries/CallMap.php#L92 so that it specifically accepts strings that are numeric

That was not removed for PHP 8 though:
https://github.com/orklah/psalm/blob/e23db18f6b1404ae8fa6035f106f5130e1774ec5/dictionaries/CallMap_80_delta.php#L2872

If you feel like scratching that itch, you should be able to just copy the `'abs\'2'` line from Callmap I linked into the removed section of the CallMap_80 to fix this
